Minio/ApiEndpoints/ObjectOperations.cs

```diff
@@ -838,9 +837,14 @@ public async Task<ObjectStat> StatObjectAsync(string bucketName, string objectNa
                     contentType = parameter.Value.ToString();
                     metaData["Content-Type"] = contentType;
                 }
-                else if (supportedHeaders.Contains(parameter.Name, StringComparer.OrdinalIgnoreCase) || parameter.Name.StartsWith("x-amz-meta-", StringComparison.OrdinalIgnoreCase))
+                else if (supportedHeaders.Contains(parameter.Name, StringComparer.OrdinalIgnoreCase))
                 {
-                    metaData[parameter.Name] = parameter.Value.ToString().ToLowerInvariant().Replace("x-amz-meta-", string.Empty);
+                    metaData[parameter.Name] = parameter.Value.ToString();
+                }
+                else if (parameter.Name.StartsWith("x-amz-meta-", StringComparison.OrdinalIgnoreCase))
+                {
+                    metaData[parameter.Name] = parameter.Value.ToString();
```

Comment:
@ngbrown , this can be removed -SDK should be stripping off the x-amz-meta- prefix from headers.

Reply:
The idea is that it wasn't doing this up until now, so not including the original header will be a breaking change.  In my application's case, I was both setting and retreiving the headers with `x-amz-meta-` already prefixed.

Reply:
I think it is still ok to break in this case. We cannot be preserving both. Like @poornas explained, SDKs should be always stripping the `x-amz-meta-` prefix.

IMO, Let this be a breaking change. Perhaps we can bump up the SDK version. cc @kannappanr

Reply:
I've removed the compatibility line and the headers are now stripped of the `x-amz-meta-` prefix.